components/upload/UploadList.tsx

```diff
@@ -64,8 +64,8 @@ export default class UploadList extends React.Component<UploadListProps, any> {
 
   // https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsDataURL
   previewFile = (file: File | Blob, callback: Function) => {
-    if (file.type && !imageTypes.includes(file.type)) {
-      callback('');
+    if (file.type.indexOf('image/') === -1) {
+      return callback('');
     }
```

Comment:
We can't change this way, it will break other correct image urls.

Reply:
How about

```
file.type && !imageTypes.includes(file.type) && file.type.indexOf('image/') === -1
```

Reply:
Thanks for answering
1. file.type can be ""
For example if you to upload something.dmg the check won't pass and FileReader will be called
2. expression (!imageTypes.includes(file.type)) always be true because file.type have MIME format https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types (image/png, video/ogg, ...) and imageTypes = ['image', 'webp', 'png', 'svg', 'gif', 'jpg', 'jpeg', 'bmp', 'dpg']
What do you think?

Reply:
You are right, could you help us fix this line too? https://github.com/ant-design/ant-design/blob/faa05a9d4da3473f32dde36638b67fd484d22fc3/components/upload/UploadList.tsx#L21

And it is better to add some test cases in case of breaking it again.

Reply:
Ok, I will do it